pkg/handlers/webhookreceiver.go

```diff
@@ -42,23 +64,157 @@ func (h *WebhookReceiverHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 		http.Error(w, "Bad request body", http.StatusBadRequest)
 		return
 	}
+
 	// process request
-	go h.processAMReceiver(alertData)
+	response := h.processAMReceiver(alertData, r.Context())
 
 	// write response
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusOK)
-	response := AMReceiverResponse{
-		Status: "ok",
-	}
+	w.WriteHeader(response.Code)
 	err = json.NewEncoder(w).Encode(response)
 	if err != nil {
 		log.Errorf("Failed to write to response: %s\n", err)
-		http.Error(w, "Failed to write to responss", http.StatusInternalServerError)
+		http.Error(w, "Failed to write to response", http.StatusInternalServerError)
 		return
 	}
 }
 
-func (h *WebhookReceiverHandler) processAMReceiver(d AMReceiverData) {
+func (h *WebhookReceiverHandler) processAMReceiver(d AMReceiverData, ctx context.Context) *AMReceiverResponse {
 	log.WithField("AMReceiverData", d).Info("Process alert data")
+
+	// Let's get all ManagedNotifications
+	mnl := &oav1alpha1.ManagedNotificationList{}
+	listOptions := []client.ListOption{
+		client.InNamespace("openshift-ocm-agent-operator"),
+	}
+	err := h.c.List(ctx, mnl, listOptions...)
+	if err != nil {
+		log.WithError(err).Error("unable to list managed notifications")
+		return &AMReceiverResponse{Error: err, Status: "unable to list managed notifications", Code: http.StatusInternalServerError}
+	}
+
+	for _, alert := range d.Alerts.Firing() {
+		// Ignore alerts which don't have a name
+		alertname, err := alertName(alert)
+		if err != nil {
+			log.WithError(err).Error("alertname missing for alert")
+			continue
+		}
+
+		// Ignore alerts which don't have the send_managed_notification label
+		if val, ok := alert.Labels[AMLabelManagedNotification]; !ok || val == "false" {
+			log.WithField("alertname", alertname).Debug("ignoring alert with no send_managed_notification label")
+			continue
+		}
+
+		// Ignore alerts which don't have a template defined
+		if _, ok := alert.Labels[AMLabelTemplateName]; !ok {
+			log.WithField("alertname", alertname).Error("alert does not have template defined")
+			continue
+		}
+
+		template, mn, err := getTemplate(alert.Labels[AMLabelTemplateName], mnl)
```

Comment:
`alert.Labels[AMLabelTemplateName]` is this a string?

Seems it returns a `(val, bool)` from line 111?

Reply:
Just did a slight refactor to make this clearer of what's going on.